tuf/ngclient/updater.py

```diff
@@ -473,7 +472,10 @@ def _preorder_depth_first_walk(
                 child_roles_to_visit.reverse()
                 delegations_to_visit.extend(child_roles_to_visit)
 
-        if number_of_delegations == 0 and len(delegations_to_visit) > 0:
+        if (
+            len(visited_role_names) > self.config.max_delegations
+            and len(delegations_to_visit) > 0
+        ):
```

Comment:
```suggestion
        if len(delegations_to_visit) > 0:
```
I like the simplification you did, and I think this now seems clear... if we have delegations to visit, we must have aborted because max delegations or do I misunderstand?

Reply:
This is correct, I've missed it.